Added Agility CMS example 🏆 (#12788)
vercel/next.js: examples/cms-agilitycms/README.md

# A statically generated blog example using Next.js and Agility CMS

This example showcases Next.js's [Static Generation](https://nextjs.org/docs/basic-features/pages) feature using [Agility CMS](https://www.agilitycms.com) as the data source.

> `IMPORTANT` - This example uses Agility CMS's [**Page Management**](https://agilitycms.com/resources/posts/page-management-in-agility-cms-vs-other-headless-cmss) features. This means that the CMS ultimately drives what pages are available and what content is on each page. This enables **Editors** to focus on managing their pages, while allowing you, (the **Developer**) to focus on building UI components for the editors to compose their pages.

## Demo

> - **Live**: [`https://next-blog-agilitycms.now.sh/](https://next-blog-agilitycms.now.sh/)

Reviewer: Sorry, forgot to remove the backtick

```suggestion
- **Live**: [https://next-blog-agilitycms.now.sh/](https://next-blog-agilitycms.now.sh/)
```

# A statically generated blog example using Next.js and Agility CMS

This example showcases Next.js's [Static Generation](https://nextjs.org/docs/basic-features/pages) feature using [Agility CMS](https://www.agilitycms.com) as the data source.

> `IMPORTANT` - This example uses Agility CMS's [**Page Management**](https://agilitycms.com/resources/posts/page-management-in-agility-cms-vs-other-headless-cmss) features. This means that the CMS ultimately drives what pages are available and what content is on each page. This enables **Editors** to focus on managing their pages, while allowing you, (the **Developer**) to focus on building UI components for the editors to compose their pages.

## Demo

> [`LIVE` - https://next-blog-agilitycms.now.sh/](https://next-blog-agilitycms.now.sh/)
> 
> [`PREVIEW MODE` - https://next-blog-agilitycms.now.sh/?agilitypreviewkey](https://next-blog-agilitycms.now.sh/?agilitypreviewkey=GzL%2fio1pLkfKc9BR1%2fC1cDQeKjL0AkwrTAJ22q3UEjAcOhyrqZejDkDv4kMlBKqrEuQxsuRyiP%2bUaykDYlJ%2fJg%3d%3d)

Reviewer: ```suggestion
- **Live**: [`https://next-blog-agilitycms.now.sh/](https://next-blog-agilitycms.now.sh/)
- **Preview Mode**: [https://next-blog-agilitycms.now.sh/?agilitypreviewkey=...](https://next-blog-agilitycms.now.sh/?agilitypreviewkey=GzL%2fio1pLkfKc9BR1%2fC1cDQeKjL0AkwrTAJ22q3UEjAcOhyrqZejDkDv4kMlBKqrEuQxsuRyiP%2bUaykDYlJ%2fJg%3d%3d)
```

### Related examples

- [Agility CMS Sample Starter](https://github.com/agility/agilitycms-next-starter-ssg)
- [Blog Starter](/examples/blog-starter)
- [Sanity](/examples/cms-sanity)
- [TakeShape](/examples/cms-takeshape)

> - [Prismic](/examples/cms-prismic)

Reviewer: Need to add Contentful, DatoCMS, Wordpress, and Strapi

## How to use

### Using `create-next-app`

Execute [`create-next-app`](https://github.com/zeit/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:

```bash
npm init next-app --example cms-agilitycms cms-agilitycms-app
# or
yarn create next-app --example cms-agilitycms cms-agilitycms-app
```

### Download manually

Download the example:

```bash
curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/cms-agilitycms
cd cms-agilitycms
```

## Configuration

### How is this Different from Other CMS Examples?

> We believe **Editors** should have full control of their pages and what content is on each page without getting into code.

Reviewer: I think we want to avoid using "We" because it might suggest that "We" = "Next.js team", which is not the case.

```suggestion
The key principle behind Agility CMS is that **Editors** should have full control of their pages and what content is on each page without getting into code.
```

- `NEXT_EXAMPLE_CMS_AGILITY_SECURITY_KEY` should be the **Security Key** field that can be found in **Settings** > **Global Security** - this is used to communicate between the CMS and your site to validate [Preview Mode](https://nextjs.org/docs/advanced-features/preview-mode)

Your `.env.local` file should look like this:

```bash
NEXT_EXAMPLE_CMS_AGILITY_GUID=...
NEXT_EXAMPLE_CMS_AGILITY_API_FETCH_KEY=...
NEXT_EXAMPLE_CMS_AGILITY_API_PREVIEW_KEY=...
NEXT_EXAMPLE_CMS_AGILITY_SECURITY_KEY=...
```

### Step 16. Run Next.js in development mode

```bash
npm install
npm run dev

# or

yarn install
yarn dev
```

Your blog should be up and running on [http://localhost:3000](http://localhost:3000)! If it doesn't work, post on [GitHub discussions](https://github.com/zeit/next.js/discussions).

### Step 17. Deploy on Vercel

You can deploy this app to the cloud with [Vercel](https://vercel.com/import?filter=next.js&utm_source=github&utm_medium=readme&utm_campaign=next-example) ([Documentation](https://nextjs.org/docs/deployment)).

> To deploy on Vercel, you need to set the environment variables using the [Vercel CLI](https://zeit.co/download) ([Documentation](https://vercel.com/docs/now-cli#commands/secrets)).

Reviewer: ```suggestion
To deploy on Vercel, you need to set the environment variables using the [Vercel CLI](https://vercel.com/download) ([Documentation](https://vercel.com/docs/now-cli#commands/secrets)).
```
